### ISSUE TITLE
0.13 is broke on iOS9 Safari/Chrome/Firefox

### ISSUE THREAD

**Issue by aligajani** [sahat/satellizer#680]
It just does not login and redirect back to the landing page.

**Comment by mikepc** [comment]
I am experiencing this as well. Can anyone can confirm an earlier version that works?

**Comment by mikepc** [comment]
I can confirm that if you remove the reverse proxy, the issue is drastically reduced, almost to acceptable levels. Is there a header or something I might need to forward?

**Comment by aligajani** [comment]
0.12 works

On Fri, Dec 18, 2015 at 9:22 AM, mikepc <notifications@github.com> wrote:

> I can confirm that if you remove the reverse proxy, the issue is
> drastically reduced, almost to acceptable levels. Is there a header or
> something I might need to forward?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sahat/satellizer/issues/680#issuecomment-165670338>.
>



-- 


Ali Gajani
Founder at Mr. Geek
www.mrgeek.me
www.aligajani.com

**Comment by mikepc** [comment]
The .13.2 seems to be stable. Keeping my eye on it, if it starts misbehaving I'm going to roll it back to .12

**Comment by mikepc** [comment]
Went back to 12.5 and it didn't work on iPhone at all, 13.2 was actually more stable :\

**Comment by aligajani** [comment]
Strange, 0.13.2 didn't work on my iPhone , but 0.12.5 does work perfectly
fine on all devices iOS 8 +.

On Sat, Dec 19, 2015 at 2:08 PM, mikepc <notifications@github.com> wrote:

> Went back to 12.5 and it didn't work on iPhone at all, 13.2 was actually
> more stable :\
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sahat/satellizer/issues/680#issuecomment-165987269>.
>



-- 


Ali Gajani
Founder at Mr. Geek
www.mrgeek.me
www.aligajani.com

**Comment by mikepc** [comment]
So actually it isn't broken for iPhone, it seems that if there's a network latency problem (eg. the oauth2 popup url isn't responding) we're not receiving errors. Satellizer is just redirecting to the home page and not reporting any errors. Trying to figure out where to trap this. I isolated it by trying different networks and I have one troublesome network, but the other two are fine, and satellizer works without a problem. So now I have to isolate how to trap this condition and at least report the error back saying there's a problem.

**Comment by mikepc** [comment]
I'm going to try 0.12.5 on my good networks

**Comment by mikepc** [comment]
Yeah it seems that on slower networks, for some reason the dialog is timing out or something along those lines.  I can't really capture it. This is happening on 0.12.5. On fast networks it's stable though. Even on the cell's LTE network.

**Comment by mikepc** [comment]
For instance, at Starbucks using starbucks Wifi it was failing > half the time, but when I disconnected from wifi it would work 100% of the time.

**Comment by aligajani** [comment]
Nice, so we know the problem is?

**Comment by sahat** [comment]
I wasn't aware of a timeout issue. Will run some tests and try to come up with a solution. As for redirecting to the homepage inside a popup, that was the simplest solution at the time of writing this library. Ideally I'd like a popup to redirect to a custom page whose only responsibility is parsing `code` or `access_token` string. It could be a blank HTML page or custom page that says "Loading...", this page is where you could also potentially display an error message if URL contains `error` query string.

This could be easily achieved by asking users of this library to include an additional file like `oauth_callback.html`, adding a route in app.js, and redirecting OAuth providers to that page  `http://localhost:3000/oauth_callback.html` instead of `http://localhost:3000`. However, I would prefer a more elegant solution, one that does not require additional files, but a self-contained solution within satellizer.js file. The only reason it has to redirect back to site root is so that satellizer.js could be loaded and check for the presence of `code` / `access_token` query param.

**Comment by aligajani** [comment]
So what should we do ideally ?

**Comment by kiro64** [comment]
+1

**Comment by xdgimf** [comment]
+1

**Comment by barryvdh** [comment]
Telling the user to create an 'oauth_callback.html'  file and link to that, would be fine with me. Better than having errors ;)

**Comment by aligajani** [comment]
Guys, any update to this?

**Comment by alicantokdemir** [comment]
I'm having same issues with ios. Android and windows phone works without a problem though. Btw, anybody can advise a good way to test this on pc?

**Comment by aligajani** [comment]
0.13.4 seems to have fixed it to a great degree +1

**Comment by kiro64** [comment]
@aligajani Can you confirm that, and did you test with Opera Mini on IOS?

**Comment by mikepc** [comment]
Satellizer is working like a champ, all the issues I faced were with my implementation.  

http://www.frobl.com to see the app live in production

**Comment by aligajani** [comment]
Same here, works nice. Althought Chrome on iOS is still a bust.

On Thu, Feb 4, 2016 at 4:35 PM, mikepc <notifications@github.com> wrote:

> Satellizer is working like a champ, all the issues I faced were with my
> implementation.
>
> http://www.frobl.com to see the app live in production
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sahat/satellizer/issues/680#issuecomment-179934293>.
>



-- 


Ali Gajani
Founder at Mr. Geek
www.mrgeek.me
www.aligajani.com

**Comment by mikepc** [comment]
Wish I had a better answer than "it works for me" but I really don't. I am able to log in with Chrome on IOS and with the droid browsers.
My advice is to look to any functionality you surrounding urls the manipulation of them, check all logic that is being executed at .config or .run() to make sure that when the popup receives the authrization token url that url isn't being manhandled

Sent from Outlook Mobile




On Thu, Feb 4, 2016 at 8:48 AM -0800, "Ali Gajani" <notifications@github.com> wrote:










Same here, works nice. Althought Chrome on iOS is still a bust.



On Thu, Feb 4, 2016 at 4:35 PM, mikepc <notifications@github.com> wrote:



> Satellizer is working like a champ, all the issues I faced were with my

> implementation.

>

> http://www.frobl.com to see the app live in production

>

> —

> Reply to this email directly or view it on GitHub

> <https://github.com/sahat/satellizer/issues/680#issuecomment-179934293>.

>







-- 





Ali Gajani

Founder at Mr. Geek

www.mrgeek.me

www.aligajani.com

**Comment by aligajani** [comment]
@mikepc  It worked on Chrome iOS 9 once I logged out and retried on Facebook login, but initially it didn't. So I would say..I don't know what could be causing this. It's definitely not my code for sure. It works fine on Safari. However, I must add, the latest update has improved the performance in terms of logging in successfully.

**Comment by pietrovismara** [comment]
Safari on iOS 9 is working flawlessly on release 0.13.3
Thanks to @mikepc for providing demonstration, you can also try to login at http://swite.com to check it.

**Comment by aligajani** [comment]
Okay coming back here once more. It's still problematic on 0.13.3 , tried on iOS Safari using 9.3.5.  Also, Pietr, your login is broken.

**Comment by ElBedeawi** [comment]
#1040 same issue here on 0.15.5
